portality/scripts/article_duplicates_report_remove.py

```diff
@@ -0,0 +1,139 @@
+    dupcount = 0
+    deleted_ids = set()
+
+    if query_override:
+        scroll_query = query_override
+    else:
+        scroll_query = {                                                                  # scroll from newest to oldest
+            "query": {"match_all": {}},
+            "sort": [{"last_updated": {"order": "desc"}}]
+        }
+
+    for a in esprit.tasks.scroll(connection, 'article', q=scroll_query):
+        article = Article(_source=a)
+
+        if article.id in deleted_ids:                               # only delete its duplicates if it isn't a duplicate
+            continue
+
+        duplicates = XWalk.discover_duplicates(article, owner)
+        if duplicates:
+            doi_dups = [d.id for d in duplicates.get('doi', [])]
+            fulltext_dups = [d.id for d in duplicates.get('fulltext', [])]
+            doi_dups.sort(), fulltext_dups.sort()                            # Sort so we can visually compare the lists
+            print "\n{0}".format(article.id)
+            if doi_dups:
+                print "\t{0} DOI duplicates: {1}".format(len(doi_dups), ", ".join(doi_dups))
+            if fulltext_dups:
+                print "\t{0} fulltext duplicates: {1}".format(len(fulltext_dups), ", ".join(fulltext_dups))
+
+            set_of_duplicates = set(doi_dups + fulltext_dups)
+            dupcount += len(set_of_duplicates) + 1                     # The detected duplicates plus the article itself
+
+            if delete:
+                for dup in set_of_duplicates:
+                    dup_art = article.pull(dup)
+                    if dup_art is not None:
+                        if snapshot:
+                            dup_art.snapshot()
+                        dup_art.delete()
+                        deleted_ids.add(dup)
+            else:
+                deleted_ids = deleted_ids | set_of_duplicates
+
+    return dupcount, len(deleted_ids)
+
+
+def duplicates_per_account(connection, delete, snapshot):
+    dupcount = 0
+    delcount = 0
+
+    for acc in esprit.tasks.scroll(connection, 'account'):
+        account = Account(**acc)
+
+        js = Journal.all(terms={"admin.owner": [account.id]})
+        if js:
+            print "Account {0}".format(account.id)
+
+            for j in js:
+                issns = j.bibjson().issns()
+                scroll_query = {
+                    "query": {
+                        "filtered": {
+                            "filter": {
+                                "bool": {
+                                    "must": [
+                                        {"terms": {"index.issn.exact": issns}}
+                                     ]
+                                }
+                            },
+                            "query": {"match_all": {}}
+                        }
+                    },
+                    "sort": [{"last_updated": {"order": "desc"}}]
+                }
```

Comment:
In fact, I think to get the issns you are after, Journal.issns_by_owner is the function you need here.

Reply:
That's right, if I'm just using the ISSNs to seed my query, it's fewer loops to get them per account rather than per journal.